linera-core/src/client/mod.rs

```diff
@@ -2825,9 +2826,12 @@ impl<Env: Environment> ChainClient<Env> {
         // Send the query to validators.
         let certificate = if round.is_fast() {
             let hashed_value = ConfirmedBlock::new(block);
-            self.client
+            let certificate = self
+                .client
                 .submit_block_proposal(&committee, proposal, hashed_value)
-                .await?
+                .await?;
+            self.update_validators(Some(&committee)).await?;
```

Comment:
For non-fast rounds we don't need to update validators explicitly since we already call `finalize_block(confirmed_certificate)`.

Reply:
Good catch! :+1: